mrjones2014/yaclt
# Matjones/79 hooks refactor

Resolves: #79 

@hsheikhali1 sorry this is kinda massive, I randomly decided to add `eslint` in this MR as well 😬

# How to Test

1. Add the following options to `yacltrc.js`:
```js
  preNew: () => console.log("!!!!! PRE NEW"),
  postNew: () => console.log("!!! POST NEW"),
  preValidate: () => console.log("!!! PRE VALIDATE"),
  postValidate: () => console.log("!!! POST VALIDATE"),
  prePrepare: () => console.log("!!! PRE PREPARE"),
  postPrepare: () => console.log("!!! POST PREPARE"),
```
2. Run `yarn build`
3. Run each command (`new`, `validate`, `prepare-release`) from `./dist/index.js` and ensure you see the appropriate `console.log` outputs. Note that since `prepare-release` also runs `validate`, the output for that one should be:
```
!!! PRE PREPARE
!!! PRE VALIDATE
✔ All changelog entries formatted correctly!
!!! POST VALIDATE
✔ CHANGELOG.md updated! Be sure to review the changes before comitting.
!!! POST PREPARE
```
4. Go back to `yacltrc.js` and update the hook functions to `return false`, verify that they now halt command execution
5. Go back to `yacltrc.js` and make all the hook functions `async`, remove the `return false`, then test step 3 again
6. Go back to `yacltrc.js` and keep the hook functions `async` but update them to `return false` again, and verify that they halt command execution
